### PR TITLE
Moves design picker after the goals step

### PR DESCRIPTION
**Pull request by p-jackson**
> [!WARNING]  
> ~DO NOT MERGE UNTIL D168479-code HAS BEEN MERGED~
> Merged via: Automattic/wpcom/pull/168479 

Related to #97208

## Proposed Changes

* Adds design picker to `onboarding`
* TODO Removes design picker from `site-setup` in https://github.com/Automattic/wp-calypso/pull/97540

https://github.com/user-attachments/assets/d84e5aa7-d1db-49b9-b34c-df291ba45a10

## Why are these changes being made?

*

## Testing Instructions

* ~Apply the patch D168479-code~
* Make sure you're logged out
* Go to /setup/onboarding, should show goals screen first, if not, please add `flags=onboarding/goals-first`
* Second step is design picker
* Select a design and see if it shows the preview

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Review by candy02058912 — COMMENTED** [review]
@p-jackson I noticed this TODO item in the PR description

> TODO Removes design picker from site-setup

I'm a bit confused on what this means?

I think this PR is good to mark as "Ready for review" as it fulfills the title of "moves design picker after the goals step."

Upcoming work:
* the broken layout that we might/might not address per the prototype
* make sure the "Continue" or "Unlock theme" CTA works

preview@2x

**Comment by zaguiini** [comment]
>make sure the "Continue" or "Unlock theme" CTA works

"Continue" works for me. "Unlock theme" doesn't, and it's because we have some site slug gating within `handleCheckout`. I think this snippet might help move us forward:

```diff
diff --git a/client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx b/client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx
index 29df82da807..782df7dd6db 100644
--- a/client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx
+++ b/client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx
@@ -532,26 +532,30 @@ const UnifiedDesignPickerStep: Step = ( { navigation, flow, stepName } ) => {
 			forceRedirection: true,
 		} );
 	}
-	function handleCheckout() {
+
+	function goToNextStep() {
+		if ( ! wpcomSiteSlug && ! siteSlug ) {
+			return pickDesign();
+		}
+
+		navigateToCheckout();
+	}
+
+	function handleCheckoutModalConfirmation() {
 		recordTracksEvent( 'calypso_signup_design_upgrade_modal_checkout_button_click', {
 			theme: selectedDesign?.slug,
 			theme_tier: selectedDesign?.design_tier,
 			is_externally_managed: selectedDesign?.is_externally_managed,
 		} );
 
-		if ( siteSlugOrId ) {
-			// We want to display the Eligibility Modal only for externally managed themes
-			// and when no domain was purchased yet.
-			if (
-				selectedDesign?.is_externally_managed &&
-				hasEligibilityMessages &&
-				! didPurchaseDomain
-			) {
-				setShowEligibility( true );
-			} else {
-				navigateToCheckout();
-			}
+		// We want to display the Eligibility Modal only for externally managed themes
+		// and when no domain was purchased yet.
+		if ( selectedDesign?.is_externally_managed && hasEligibilityMessages && ! didPurchaseDomain ) {
+			setShowEligibility( true );
+		} else {
+			goToNextStep();
 		}
+
 		setShowUpgradeModal( false );
 	}
 
@@ -698,6 +702,7 @@ const UnifiedDesignPickerStep: Step = ( { navigation, flow, stepName } ) => {
 				{
 					selectedDesign: _selectedDesign,
 					selectedSiteCategory: categorization.selections?.join( ',' ),
+					plan: requiredPlanSlug,
 				},
 				{ ...( positionIndex >= 0 && { position_index: positionIndex } ) }
 			);
@@ -833,7 +838,7 @@ const UnifiedDesignPickerStep: Step = ( { navigation, flow, stepName } ) => {
 						marketplaceProduct={ selectedMarketplaceProduct }
 						requiredPlan={ requiredPlanSlug }
 						closeModal={ closeUpgradeModal }
-						checkout={ handleCheckout }
+						checkout={ handleCheckoutModalConfirmation }
 					/>
 				) }
 				<QueryEligibility siteId={ site?.ID } />
@@ -849,7 +854,7 @@ const UnifiedDesignPickerStep: Step = ( { navigation, flow, stepName } ) => {
 						setShowEligibility( false );
 					} }
 					handleContinue={ () => {
-						navigateToCheckout();
+						goToNextStep();
 						setShowEligibility( false );
 					} }
 				/>
diff --git a/client/landing/stepper/declarative-flow/onboarding.ts b/client/landing/stepper/declarative-flow/onboarding.ts
index 313dec5cea8..0d297d2ea3c 100644
--- a/client/landing/stepper/declarative-flow/onboarding.ts
+++ b/client/landing/stepper/declarative-flow/onboarding.ts
@@ -128,7 +128,12 @@ const onboarding: Flow = {
 				}
 
 				case 'designSetup': {
-					const { selectedDesign: _selectedDesign } = providedDependencies;
+					const { selectedDesign: _selectedDesign, plan } = providedDependencies;
+
+					if ( plan ) {
+						setPlanCartItem( plan );
+					}
+
 					if ( isAssemblerDesign( _selectedDesign as Design ) && isAssemblerSupported() ) {
 						return navigate( 'pattern-assembler' );
 					}

```

**Comment by p-jackson** [comment]
> > TODO Removes design picker from site-setup
> I'm a bit confused on what this means?

@candy02058912 it just means that the current state of the PR means users see the design picker twice. Once before account creation, then as they continue through the flow, after site creation and checkout, they see the design picker again.

They should only see it once: before account creation.

**Comment by candy02058912** [comment]
@zaguiini tested those changes, and it unblocks the theme selection; I was wondering if we'd like to add the snippet to a separate PR and set it to merge to this branch so people can check out to that branch and review it more easily if we don't want to make this PR too large.

**Comment by zaguiini** [comment]
I think that's a more sensible decision @candy02058912 since these two are related but my snippet isn't complete, so we'll definitely require more code to make it fully functional.

**Comment by zaguiini** [comment]
>I think that's a more sensible decision @candy02058912 since these two are related but I haven't thoroughly tested my snippet, so we'll probably want to give it more attention.

Scratch that. See p1734385797588649/1734385074.248909-slack-C085HCWCEDN.

>I'm getting redirected to /setup/site-setup/designSetup?siteSlug=%s&goals-at-front-experiment=true&skippedCheckout=1&categories=blog after picking a design and a domain - this is the URL that the checkout is sending me to. Not sure if it's related.

Additionally, to the snippet shared above, I tried setting the destination as `/home/%s` within the `processing` step submission in the `/setup/onboarding` flow like this:

image

But it didn't work out. After checkout, it redirected me to my site without my theme/style variation. I was talking with @p-jackson about this, and we'll need to briefly redirect to `/setup/site-setup` (or somewhere else) so that the chosen theme and variation are applied to the new site.

Considering the user already has a site by the time we call the `processing` step, an option is to store the theme and variation as an option and assign their choices to the site after checkout.

**Comment by autumnfjeld** [comment]
Pinging @arthur791004  in case you want to pick this up.

File: client/landing/stepper/declarative-flow/onboarding.ts [hunk 1]
```diff
@@ -128,11 +129,20 @@ const onboarding: Flow = {
...
-							return navigate( 'domains' );
+							return navigate( 'designSetup' );
 						}
 					}
 				}
 
+				case 'designSetup': {
+					const { selectedDesign: _selectedDesign } = providedDependencies;
+					if ( isAssemblerDesign( _selectedDesign as Design ) && isAssemblerSupported() ) {
+						return navigate( 'pattern-assembler' );
```
Comment (arthur791004) [inline client/landing/stepper/declarative-flow/onboarding.ts]: The pattern assembler has been sunset

Comment (candy02058912) [reply client/landing/stepper/declarative-flow/onboarding.ts]: Removed it 👍

**Review by candy02058912 — APPROVED** [review]
This PR is good to merge as it isn't reflected in production, and we'll follow up with https://github.com/Automattic/wp-calypso/pull/97540

**Pull request merged**